src/visualization/types/Graph/view.tsx

```diff
@@ -223,23 +195,55 @@ const XYPlot: FC<Props> = ({properties, result, timeRange, annotations}) => {
     ],
   }
 
-  if (isFlagEnabled('annotations') && annotations) {
+  if (isFlagEnabled('annotations') && annotationsModeIsActive) {
+    const createAnnotation = userModifiedAnnotation => {
+      const {message, startTime} = userModifiedAnnotation
+      dispatch(
+        writeThenFetchAndSetAnnotations([
+          {
+            summary: message,
+            startTime: new Date(startTime).getTime(),
+            endTime: new Date(startTime).getTime(),
+          },
+        ])
+      )
+    }
+
+    const doubleClickHandler = (
+      plotInteraction: InteractionHandlerArguments
+    ) => {
+      dispatch(
+        showOverlay(
+          'add-annotation',
+          {
+            createAnnotation,
+            startTime: plotInteraction.valueX,
+          },
+          dismissOverlay
+        )
+      )
+    }
+
+    config.interactionHandlers = {
+      doubleClick: doubleClickHandler,
+    }
+
     // everything is under the 'default' category for now:
-    const actualAnnotations: any[] = annotations.default ?? null
+    const selectedAnnotations: any[] = annotations?.default ?? []
```

Comment:
combined the check above (`if (isFlagEnabled('annotations') && annotations)`) into this `annotations?.`

Reply:
the clause in the if statement (starting on line 198) is really long.  it is almost 60 lines; more than fits on the screen (I had to paste it into an editor to check it to understand the code).  please put it in its own method for clarity.  

just pass the config object to the method, and the method can adjust the config object; and it will be much clearer.

Reply:
@jrenee42 i thought a bit about this - I don't want to pull the whole thing out into a function because then it's just moving the big block of code and making the reader go read it somewhere else.

i could put only the important parts inside the flag:

```js
if (flagIsEnabled('annotations') {
  config.interactionHandlers = {
    doubleClick: doubleClickHandler,
  }

  config.layers.push(annotationLayer)
}
```
and pull everything else out of the feature flag check, but then since it's a function component that renders a graph on a hot rendering path, it's going to be constantly re-creating methods it won't use.

i agree that this isn't ideal and that the huge length of this block is a smell that this should be refactored. i'm not completely clear on what needs to be refactored yet and i'm loath to do it too early and do the wrong thing. i think what i'd like to do in a separate step is pull out the functions that aren't required to be in the rendering path and move them higher up the chain. but like i said, i'm not 100% clear on the best path forward and i'd rather wait to do this until the best way forward is a little more clear.

what do you think about me writing a ticket to address this and then adding a comment linking to that ticket? so that we can address it after we get a chance to mess with the annotations a bit more?

Reply:
i think the above is a good template;  make a method to make the doubleclick handler,
and a method to make the annotationlayer.  our code should be readable/clear when it is first published, if we put refactoring into technical debt it may never happen.